### PR TITLE
Docs for examples

### PR DESCRIPTION
**Pull request by TatriX**
I'm going through the examples to better learn the framework. When there are more then 5 examples it's nice to have an overview of all of them, even better with some order of increasing complexity.

### CODE REVIEW

**Comment by TatriX** [comment]
BTW, do you think these:
```
// ------ ------
//     Model
// ------ ------
```
type of comments are actually useful? I assume that this is inspired by Elm, but still, I can't see the point of having them.

**Comment by David-OConnor** [comment]
Solid addition and changes. You're correct that the `Model`, `View` etc comments in examples are inspired by Elm and its guide. I'm 50/50 on whether they're useful, or clutter. Perhaps we should keep them in the quickstart repo and example, but not the other examples. @MartinKavik  - what do you think? We can do another PR to remove them if you'd like.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### [Update from JS](update_from_js)
+How to trigger `update` function from Javascript world.
+You'll also see how to call JS functions from Rust.
+
+### [UserMedia](user_media)
+How to show your webcam output in `video` element.
+
+### [Websocket Chat](websocket)
+Example of communicating with a server using Websockets.
```
Comment (MartinKavik) [inline examples/README.md]: I think it should be in the _Server_ section.

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+### [Mathjax](mathjax)
+An example using Mathjax, for LaTeX-like math notation.
+
+### [Orders](orders)
+How to perform commands and send messages from `update` function.
+And how to use [gloo](https://github.com/rustwasm/gloo) timers.
+
+### [Todo MVC](todomvc)
+Classic TodoMVC  example.
```
Comment (MartinKavik) [inline examples/README.md]: Extra space before `example`.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### [Integration](server_integration)
+Example of a workspace with [Actix](https://actix.rs/) server.
+
+### [Interaction](server_interaction)
+Example of communicating with a server using HTTP requests.
+
+### [Detailed interaction](server_interaction_detailed)
+A more complicated example of retrieving and using data from a server,
+ and outputing multiple wasm files.
```
Comment (MartinKavik) [inline examples/README.md]: Extra space before `and`.

File: examples/animation_frame/src/lib.rs [hunk 4]
```diff
@@ -93,14 +91,14 @@ fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
 // View
 
 fn view(model: &Model) -> Node<Msg> {
-    // scene container + sky
+    // scene container, aslo represents sky
```
Comment (MartinKavik) [inline examples/animation_frame/src/lib.rs]: typo `aslo`

File: examples/animation_frame/src/lib.rs [hunk 5]
```diff
@@ -27,7 +25,7 @@ impl Car {
 
     fn generate_color() -> CarColor {
         let hue = thread_rng().gen_range(0, 360);
-        format!("hsl({},100%,50%)", hue)
+        format!("hsl({}, 80%, 50%)", hue)
```
Comment (MartinKavik) [inline examples/animation_frame/src/lib.rs]: Is 100% too unrealistic for the car color? 😄

File: examples/websocket/src/client.rs [hunk 6]
```diff
@@ -1,8 +1,5 @@
-#[macro_use]
-extern crate seed;
-
 use js_sys::Function;
-use seed::{prelude::*, App};
+use seed::{prelude::*, App, *};
```
Comment (MartinKavik) [inline examples/websocket/src/client.rs]: `App` should be in `prelude`.

**Comment by MartinKavik** [comment]
```rust
// ------ ------
//     Model
// ------ ------
```
I use it to visually group main functions (`view`, `update`, etc.) with their data (`Msg`, `GMsg`) and helpers. At least `examples/app_builder` and real apps are more readable with them imho.

**Pull request merged**